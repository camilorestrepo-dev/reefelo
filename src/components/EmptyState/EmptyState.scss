@import "../../styles/shared.scss";

.empty-state-container {
  display: flex;
  flex-direction: column;
  align-items: center;
  margin-top: $spacing-20;

  .empty-state-image {
    width: 200px;
    height: 200px;
  }

  h2 {
    margin-bottom: $spacing-8;
  }

  p {
    margin-top: 0;
  }
}