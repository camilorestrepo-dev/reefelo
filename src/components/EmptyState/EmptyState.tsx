import "./EmptyState.scss";
import { useTranslation } from "react-i18next";
import Button from "../Button/Button";
import emptyImage from "../../assets/images/empty.svg";

type EmptyStateProps = {
  buttonText?: string;
  onRetry?: VoidFunction;
};

const EmptyState: React.FC<EmptyStateProps> = ({ buttonText, onRetry }: EmptyStateProps) => {
  const { t } = useTranslation();
  return (
    <div className="empty-state-container">
      <img src={emptyImage} className="empty-state-image" />
      <h2>{t("emptyState.title")}</h2>
      <p>{t("emptyState.subtitle")}</p>
      <Button variant="outline" className="retry" onClick={onRetry}>
        {buttonText || t("tryAgain")}
      </Button>
    </div>
  );
};

export default EmptyState;
