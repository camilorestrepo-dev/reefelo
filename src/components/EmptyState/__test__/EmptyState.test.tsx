import { render } from "@testing-library/react";
import EmptyState from "../EmptyState";

test("render correctly", () => {
  const component = render(<EmptyState />);
  expect(component.container).toMatchSnapshot();
});
