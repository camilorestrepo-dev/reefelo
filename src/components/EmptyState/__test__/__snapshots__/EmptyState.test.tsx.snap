// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
<div>
  <div
    class="empty-state-container"
  >
    <img
      class="empty-state-image"
      src="function SvgEmpty(props) {
  return React.createElement(
    'svg', 
    Object.assign({}, props, {'data-file-name': SvgEmpty.name})
  );
}"
    />
    <h2>
      emptyState.title
    </h2>
    <p>
      emptyState.subtitle
    </p>
    <button
      class="custom-btn outline-btn retry"
      type="button"
    >
      tryAgain
    </button>
  </div>
</div>
`;
